dashboard/app.py

```diff
@@ -15,20 +18,25 @@ def get_header_metrics() -> None:
         st.metric("Total Number of Plants :herb:",
                   len(selected_plants))
     with head_cols[1]:
-        st.metric("Average Soil Moisture: :potted_plant:", )
+        st.metric("Average Soil Moisture: :potted_plant:",
+                  get_average_soil_moisture(plants[name_in_selected_plants]))
 
 
 if __name__ == "__main__":
 
     load_dotenv()
 
-    plants = pd.read_csv("./../pipeline/transformed_test.csv")
+    conn = get_database_connection()
+
+    plants = load_all_plant_data(conn)
 
     st.title("Plant Sensors Dashboard")
 
     selected_plants = st.sidebar.multiselect("Selected Plants",
-                                             plants["plant_id"].unique(),
-                                             default=plants["plant_id"].unique())
+                                             plants["plant_name"].unique(),
+                                             default=plants["plant_name"].unique())
+
+    name_in_selected_plants = plants["plant_name"].isin(selected_plants)
```

Comment:
Perhaps this could be extracted to a function just to make main look a tiny bit cleaner